File: cakk-api/src/main/java/com/cakk/api/dto/request/user/UserSignInRequest.java
package com.cakk.api.dto.request.user;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;

import com.cakk.common.enums.Provider;

public record UserSignInRequest(
	@NotNull
	Provider provider,
	@NotBlank
	String idToken
) {
}

File: cakk-api/src/main/java/com/cakk/api/dto/request/user/UserSignInRequest.kt
package com.cakk.api.dto.request.user

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull

import com.cakk.common.enums.Provider

data class UserSignInRequest(
	@field:NotNull
    val provider: Provider?,
	@field:NotBlank
    val idToken: String?
)
